### PR TITLE
UI: Add warning to Custom FFmpeg mode

### PR DESCRIPTION
**Pull request by gxalpha**
### Description

Screenshot of Custom Output panel with the warning visible

### Motivation and Context

Custom FFmpeg is considered mostly unsupported by us, but we don't really communicate that anywhere in the program.

### How Has This Been Tested?

macOS 14
See the screenshot above.

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by RytoEX — CHANGES_REQUESTED** [review]
cc @Warchamp7 

Rationale:
Saying it's "intended for professionals" has two potential outcomes I want to avoid:
1. Users reacting with "I'm a professional" when what we meant was "intended for people who understand how to use FFmpeg".
2. Users may have a negative reaction to the implication that they are not professionals.

We can simply be direct and point out that the mode comes with no support. Saying it's "generally not recommended for most users" has the same pitfalls as "intended for professionals" ("I'm not most users"), and I can't think of a situation where we would ever *recommend* this mode (we might concede that it's the only way to do a specific task, but probably not recommend).

File: UI/data/locale/en-US.ini [hunk 1]
```diff
@@ -1058,6 +1058,7 @@ Basic.Settings.Output.Adv.Recording.Type.FFmpegOutput="Custom Output (FFmpeg)"
 Basic.Settings.Output.Adv.Recording.UseStreamEncoder="(Use stream encoder)"
 Basic.Settings.Output.Adv.Recording.Filename="Filename Formatting"
 Basic.Settings.Output.Adv.Recording.OverwriteIfExists="Overwrite if file exists"
+Basic.Settings.Output.Adv.FFmpeg.CustomModeWarning="Custom Output (FFmpeg) is intended for professionals and generally not recommended for most users.\nWe recommend using \"Standard\" recording type."
```
[the remark this answers is not in the document]
Comment (gxalpha) [reply UI/data/locale/en-US.ini]: Changed.

**Review by RytoEX — APPROVED** [review]
Seems okay to me.

File: UI/forms/OBSBasicSettings.ui [hunk 2]
```diff
@@ -3410,7 +3410,20 @@
                                <property name="bottomMargin">
                                 <number>9</number>
                                </property>
-                               <item row="0" column="0">
+                               <item row="0" column="1">
```
Comment (Warchamp7) [inline UI/forms/OBSBasicSettings.ui:3413]: Let's do `column="0"` and `colspan="2"`

This warning is already placed differently than other settings warnings and I'm fine making it stand out even more this way

**Review by Warchamp7 — CHANGES_REQUESTED** [review]

File: UI/forms/OBSBasicSettings.ui (hunk 2 above)
Comment (gxalpha) [reply UI/forms/OBSBasicSettings.ui:3413]: Done.
New screenshot:
Bildschirmfoto 2024-07-26 um 22 42 05

File: UI/data/locale/en-US.ini [hunk 3]
```diff
@@ -1086,6 +1086,7 @@ Basic.Settings.Output.Adv.Recording.Type.FFmpegOutput="Custom Output (FFmpeg)"
 Basic.Settings.Output.Adv.Recording.UseStreamEncoder="(Use stream encoder)"
 Basic.Settings.Output.Adv.Recording.Filename="Filename Formatting"
 Basic.Settings.Output.Adv.Recording.OverwriteIfExists="Overwrite if file exists"
+Basic.Settings.Output.Adv.FFmpeg.CustomModeWarning="Custom Output (FFmpeg) is provided with no safeguards.\nConsider using \"Standard\" recording type."
```
Comment (WizardCM) [inline UI/data/locale/en-US.ini:1089]:
```suggestion
Basic.Settings.Output.Adv.FFmpeg.CustomModeWarning="Custom Output (FFmpeg) is provided with no safeguards.\nConsider using the \"Standard\" recording type."
```

Comment (gxalpha) [reply UI/data/locale/en-US.ini]: Changed.

**Comment by gxalpha** [comment]
Warchamp approved off-thread.

**Pull request merged**